### ISSUE TITLE
kernel 5.16.x compiling error

### ISSUE THREAD

**Issue by antonellocaroli** [bondagit/aes67-linux-daemon#64]
Hi,

I get an error when I try to compile the module on kernel 5.16.x

`ravenna/ravenna-alsa-lkm/driver/RTP_audio_stream.c: In function ‘PrepareBufferLives’:
/ravenna/ravenna-alsa-lkm/driver/RTP_audio_stream.c:1044:41: warning: unused variable ‘delta’ [-Wunused-variable]
 1044 |                                 int64_t delta = self->m_tRTPStream.m_ui64LastAudioSampleReceivedSAC - pManager->get_global_SAC(pManager->user);
      |                                         ^~~~~
  CC [M]  /ravenna/ravenna-alsa-lkm/driver/RTP_stream.o
  CC [M]  /ravenna/ravenna-alsa-lkm/driver/RTP_stream_info.o
  CC [M]  /ravenna/ravenna-alsa-lkm/driver/MTAL_EthUtils.o
/ravenna/ravenna-alsa-lkm/driver/MTAL_EthUtils.c: In function ‘MTAL_DumpID’:
/ravenna/ravenna-alsa-lkm/driver/MTAL_EthUtils.c:107:25: warning: unused variable ‘by’ [-Wunused-variable]
  107 |                 uint8_t by = (ullID >> (ui32 * 8)) & 0xFF;
      |                         ^~
/ravenna/ravenna-alsa-lkm/driver/MTAL_EthUtils.c: In function ‘MTAL_DumpAppleMIDI’:
/ravenna/ravenna-alsa-lkm/driver/MTAL_EthUtils.c:556:50: warning: unused variable ‘pSynch’ [-Wunused-variable]
  556 |                         TAppleMIDI_Synch_Packet* pSynch = (TAppleMIDI_Synch_Packet*)pAppleMIDI_PacketBase;
      |                                                  ^~~~~~
/ravenna/ravenna-alsa-lkm/driver/MTAL_EthUtils.c:580:56: warning: unused variable ‘pAppleMIDI_IN_NO_OK_BY_Packet’ [-Wunused-variable]
  580 |                         TAppleMIDI_IN_NO_OK_BY_Packet* pAppleMIDI_IN_NO_OK_BY_Packet = (TAppleMIDI_IN_NO_OK_BY_Packet*)pAppleMIDI_PacketBase;
      |                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  CC [M]  /ravenna/ravenna-alsa-lkm/driver/MTAL_LKernelAPI.o
/ravenna/ravenna-alsa-lkm/driver/MTAL_LKernelAPI.c:94:10: fatal error: stdarg.h: No such file or directory
   94 | #include <stdarg.h>
      |          ^~~~~~~~~~
compilation terminated.
make[2]: *** [scripts/Makefile.build:287: /ravenna/ravenna-alsa-lkm/driver/MTAL_LKernelAPI.o] Error 1
make[1]: *** [Makefile:1846: /ravenna/ravenna-alsa-lkm/driver] Error 2
make[1]: Leaving directory '/usr/src/linux-5.16.1-rt16'
make: *** [Makefile:14: modules] Error 2`

**Comment by bondagit** [comment]
the problem doesn't seem to be related to the driver code: <stdarg.h> should be provided by your toolchain.
Try with:  https://github.com/lwfinger/rtl8723bu/issues/106

**Comment by antonellocaroli** [comment]
No, unfortunately, it is not a problem with the toochain.
with kernel 5.15.x it compiles without problems....

**Comment by andreweggleston** [comment]
This has got something to do with include paths changing on his release. I could build fine by replacing any `#include <stdarg.h>` with `#include <linux/stdarg.h>`

**Comment by antonellocaroli** [comment]
@bondagit the advice of @andreweggleston seems to work

I modified the file **MTAL_LKernelAPI.c**

```
#include <linux/stdarg.h>
int MTAL_LK_print(const char *fmt, ...)
{
    va_list args;
    int res = 0;
    va_start(args, fmt);
    res = vprintk(fmt, args);
    va_end(args);
    return res;
}
```

```
make -C /lib/modules/5.16.0-pf2-x86_64-GP-PF-ST-nvidia/build/ M=/tmp/ravenna-alsa-lkm/driver modules
make[1]: ingresso nella directory «/usr/src/linux-5.16_p2-pf»
warning: the compiler differs from the one used to build the kernel
  The kernel was built by: x86_64-pc-linux-gnu-gcc (Gentoo 11.2.1_p20220115 p4) 11.2.1 20220115
  You are using:           gcc (Gentoo 11.2.1_p20220115 p4) 11.2.1 20220115
  CC [M]  /tmp/ravenna-alsa-lkm/driver/c_wrapper_lib.o
  CC [M]  /tmp/ravenna-alsa-lkm/driver/audio_driver.o
  CC [M]  /tmp/ravenna-alsa-lkm/driver/manager.o
  CC [M]  /tmp/ravenna-alsa-lkm/driver/module_main.o
  CC [M]  /tmp/ravenna-alsa-lkm/driver/PTP.o
  CC [M]  /tmp/ravenna-alsa-lkm/driver/module_interface.o
  CC [M]  /tmp/ravenna-alsa-lkm/driver/module_netlink.o
  CC [M]  /tmp/ravenna-alsa-lkm/driver/module_timer.o
  CC [M]  /tmp/ravenna-alsa-lkm/driver/EtherTubeNetfilter.o
  CC [M]  /tmp/ravenna-alsa-lkm/driver/RTP_streams_manager.o
  CC [M]  /tmp/ravenna-alsa-lkm/driver/RTP_audio_stream.o
/tmp/ravenna-alsa-lkm/driver/RTP_audio_stream.c: In function ‘PrepareBufferLives’:
/tmp/ravenna-alsa-lkm/driver/RTP_audio_stream.c:1044:41: warning: unused variable ‘delta’ [-Wunused-variable]
 1044 |                                 int64_t delta = self->m_tRTPStream.m_ui64LastAudioSampleReceivedSAC - pManager->get_global_SAC(pManager->user);
      |                                         ^~~~~
  CC [M]  /tmp/ravenna-alsa-lkm/driver/RTP_stream.o
  CC [M]  /tmp/ravenna-alsa-lkm/driver/RTP_stream_info.o
  CC [M]  /tmp/ravenna-alsa-lkm/driver/MTAL_EthUtils.o
/tmp/ravenna-alsa-lkm/driver/MTAL_EthUtils.c: In function ‘MTAL_DumpID’:
/tmp/ravenna-alsa-lkm/driver/MTAL_EthUtils.c:107:25: warning: unused variable ‘by’ [-Wunused-variable]
  107 |                 uint8_t by = (ullID >> (ui32 * 8)) & 0xFF;
      |                         ^~
/tmp/ravenna-alsa-lkm/driver/MTAL_EthUtils.c: In function ‘MTAL_DumpAppleMIDI’:
/tmp/ravenna-alsa-lkm/driver/MTAL_EthUtils.c:556:50: warning: unused variable ‘pSynch’ [-Wunused-variable]
  556 |                         TAppleMIDI_Synch_Packet* pSynch = (TAppleMIDI_Synch_Packet*)pAppleMIDI_PacketBase;
      |                                                  ^~~~~~
/tmp/ravenna-alsa-lkm/driver/MTAL_EthUtils.c:580:56: warning: unused variable ‘pAppleMIDI_IN_NO_OK_BY_Packet’ [-Wunused-variable]
  580 |                         TAppleMIDI_IN_NO_OK_BY_Packet* pAppleMIDI_IN_NO_OK_BY_Packet = (TAppleMIDI_IN_NO_OK_BY_Packet*)pAppleMIDI_PacketBase;
      |                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  CC [M]  /tmp/ravenna-alsa-lkm/driver/MTAL_LKernelAPI.o
  CC [M]  /tmp/ravenna-alsa-lkm/driver/MTConvert.o
  LD [M]  /tmp/ravenna-alsa-lkm/driver/MergingRavennaALSA.o
  MODPOST /tmp/ravenna-alsa-lkm/driver/Module.symvers
  CC [M]  /tmp/ravenna-alsa-lkm/driver/MergingRavennaALSA.mod.o
  LD [M]  /tmp/ravenna-alsa-lkm/driver/MergingRavennaALSA.ko
make[1]: uscita dalla directory «/usr/src/linux-5.16_p2-pf»
```

**Comment by bondagit** [comment]
This looks like a problem with kernel 5.16.x only.
You can just create a symlink stdarg.h -> linux/stdarg.h in your kernel include directory.

**Comment by andreweggleston** [comment]
https://github.com/torvalds/linux/commit/c0891ac15f0428ffa81b2e818d416bdf3cb74ab6

linux 5.15 and forward looks to ship with its own stdarg from gcc4.2.

**Comment by bondagit** [comment]
should be be fixed by bondagit/ravenna-alsa-lkm@ce46ebda
